cadmiumcat/books-api
# Adds mongodb

### What
- I've replaced the in-memory storage with mongodb and refactored... a few things
- Added mock data store for tests
- Replaced `nil` context with actual context `ctx`

### How to review
- `make convey` to check the tests pass
- Try the request to add a book. For example:
`{
    "Title":"Girl, Woman, Other",
    "Author":"Bernardine Evaristo"
}`

- Check that you can list all books and a book with specific id

#### Things I'm unsure about: 
- Does the structure make sense? i.e. folder structure and responsibilities

### Who can review
Anyone
